src/HookTrait.php

```diff
@@ -18,32 +18,37 @@ trait HookTrait
      */
     protected $hooks = [];
 
+    /**
+     * Next hook index counter.
+     *
+     * @var int
+     */
+    private $_hookIndexCounter = 0;
+
     /**
      * @deprecated use onHook instead
      */
-    public function addHook($spot, $fx, $args = null, $priority = null)
+    public function addHook($spot, $fx, array $args = null, int $priority = null)
     {
-        return $this->onHook(...func_get_args());
+        throw new Exception('Method addHook is deprecated, use onHook instead.');
```

Comment:
This is to big backward compatibility break and should be done only on next major release.
For now I think addHook should stay as proxy method for compatibility.

Reply:
@DarkSide666 Yes, it seems so, I will remove that exception.

What about introducing some static method that can be easily called in deprecated code to emit deprecated warning/php notice? Otherwise almost all people ignore deprecations until the deprecated feature is removed.

Reply:
Yes that could be useful I guess.